config/v1/types_authentication.go

```diff
@@ -771,3 +826,46 @@ type TokenRequiredClaim struct {
 	// +required
 	RequiredValue string `json:"requiredValue"`
 }
+
+type TokenClaimValidationCELRule struct {
+	// expression is a CEL expression evaluated against token claims.
+	// The expression must be a non-empty string and no longer than 1024 characters.
+	// The expression must return a boolean value where 'true' signals a valid token and 'false' an invalid one.
+	// This field is required.
```

Comment:
Nit: wording consistency for constraints across fields:
```suggestion
	// expression is required, must be at least 1 character in length and must not exceed 1024 characters.
	// The expression must return a boolean value where 'true' signals a valid token and 'false' an invalid one.
```